api/server.go

```diff
@@ -0,0 +1,173 @@
+		return
+	}
+	if err = lt.Run(); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	writeJsonResponse(w, map[string]string{"message": "load test started"})
+}
+
+func (a *API) stopLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	if err = lt.Stop(); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	writeJsonResponse(w, map[string]string{"message": "load test stopped"})
+}
+
+func (a *API) destroyLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+
+	_ = lt.Stop() // we are ignoring the error here in case the load test was previously stopped
+
+	delete(a.loadTests, mux.Vars(r)["id"])
+	writeJsonResponse(w, map[string]string{"message": "load test destroyed"})
+}
+
+func (a *API) getLoadTestStatusHandler(w http.ResponseWriter, r *http.Request) {
+	_, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(map[string]string{"status": "Not yet implemented"})
+}
+
+func getAmount(r *http.Request) int {
+	amountStr := r.FormValue("amount")
+	amount := 1
+
+	if amountStr != "" {
+		if a, err := strconv.ParseInt(amountStr, 10, 16); err == nil && a > 0 {
+			amount = int(a)
+		}
+	}
+	return amount
+}
+
+func (a *API) addUserHandler(w http.ResponseWriter, r *http.Request) {
+	amount := getAmount(r)
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+
+	for i := 0; i < amount; i++ {
+		if err = lt.AddUser(); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	writeJsonResponse(w, map[string]string{"message": "user added"})
+}
+
+func (a *API) removeUserHandler(w http.ResponseWriter, r *http.Request) {
+	amount := getAmount(r)
+
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+
+	for i := 0; i < amount; i++ {
+		if err = lt.RemoveUser(); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+	writeJsonResponse(w, map[string]string{"message": "user removed"})
+}
+
+func SetupAPIRouter() *mux.Router {
+	router := mux.NewRouter()
+	r := router.PathPrefix("/loadtest").Subrouter()
+
+	agent := API{loadTests: make(map[string]*loadtest.LoadTester)}
+	r.HandleFunc("/create", agent.createLoadTestHandler).Methods("POST")
+	r.HandleFunc("/run/{id}", agent.runLoadTestHandler).Methods("POST")
+	r.HandleFunc("/stop/{id}", agent.stopLoadTestHandler).Methods("POST")
+	r.HandleFunc("/destroy/{id}", agent.destroyLoadTestHandler).Methods("POST")
+	r.HandleFunc("/status/{id}", agent.getLoadTestStatusHandler).Methods("GET")
+	r.HandleFunc("/user/{id}", agent.addUserHandler).Methods("PUT").Queries("amount", "{[0-9]*?}")
+	r.HandleFunc("/user/{id}", agent.removeUserHandler).Methods("DELETE").Queries("amount", "{[0-9]*?}")
```

Comment:
I think `destroy` is the only call where it makes sense to use `DELETE` since we are actually deleting an object.
For adding/removing users I believe `POST` is more precise. Thoughts?